jdaviz/configs/imviz/tests/utils.py

```diff
@@ -201,3 +201,111 @@ def setup_class(self, imviz_helper):
         # Since we are not really displaying, need this to test zoom.
         self.viewer.shape = (100, 100)
         self.viewer.state._set_axes_aspect_ratio(1)
+
+
+# --- This was used to generate data/roman_wfi_image_model.asdf ---
+
+def create_example_gwcs(shape):
+    # Example adapted from photutils:
+    #   https://github.com/astropy/photutils/blob/
+    #   2825356f1d876cacefb3a03d104a4c563065375f/photutils/datasets/make.py#L821
+    rho = np.pi / 3.0
+    # Roman plate scale:
+    scale = (0.11 * (u.arcsec / u.pixel)).to_value(u.deg / u.pixel)
+
+    shift_by_crpix = (models.Shift((-shape[1] / 2) + 1)
+                      & models.Shift((-shape[0] / 2) + 1))
+
+    cd_matrix = np.array([[-scale * np.cos(rho), scale * np.sin(rho)],
+                          [scale * np.sin(rho), scale * np.cos(rho)]])
+
+    rotation = models.AffineTransformation2D(cd_matrix, translation=[0, 0])
+    rotation.inverse = models.AffineTransformation2D(
+        np.linalg.inv(cd_matrix), translation=[0, 0])
+
+    tan = models.Pix2Sky_TAN()
+    celestial_rotation = models.RotateNative2Celestial(197.8925, -1.36555556, 180.0)
+
+    det2sky = shift_by_crpix | rotation | tan | celestial_rotation
+    det2sky.name = 'linear_transform'
+
+    detector_frame = cf.Frame2D(name='detector', axes_names=('x', 'y'), unit=(u.pix, u.pix))
+
+    sky_frame = cf.CelestialFrame(reference_frame=ICRS(), name='icrs', unit=(u.deg, u.deg))
+
+    pipeline = [(detector_frame, det2sky), (sky_frame, None)]
+
+    return gwcs_wcs.WCS(pipeline)
+
+
+def create_wfi_image_model(image_shape, **kwargs):
+    """
+    Create a dummy Roman WFI Image datamodel instance with valid values
+    for attributes required by the schema.
+
+    Requires roman_datamodels >= 0.14.2
+
+    Adapted from:https://github.com/spacetelescope/roman_datamodels/blob/932afed416e02bf0e684973f03548eeefe973b9e/src/roman_datamodels/testing/factories.py#L1224
```

Comment:
why cant the factory function just be imported from roman_datamodels? i worry this would be hard to keep up to date as the files evolve

Reply:
Same comment as https://github.com/spacetelescope/jdaviz/pull/1822#pullrequestreview-1385545865 or is this something different?

Reply:
The function in `roman_datamodels` uses fixed pixel dimensions, which match the detector shape of Roman. For testing, it's convenient to make smaller arrays.

Reply:
p.s. LoL I was blind. I saw this URL but didn't cross my mind that we should just use the upstream version. Good catch!

Reply:
To follow up: the link in the docstring corresponds to the version that I modified, and you can see that the detector shape can't be modified. On main, there's [a corresponding function that's more flexible](https://github.com/spacetelescope/roman_datamodels/blob/626e0da697614cefaef3b26d06fc3f528e2c7dab/src/roman_datamodels/maker_utils.py#L485)! I'll probably switch to using it instead.

Reply:
Turns out this was an easy switch, see bd0ef9c.